### DISCUSSION TITLE
Failed to download resource "theora"

### DISCUSSION THREAD

**Discussion by pyrolusite-d** [Homebrew/discussions#2357] [Writing Formulae/Casks]
While installing Allegro with `brew install allegro` the process keeps being cut off by brew failing to download resource "theora."

==> Downloading https://downloads.xiph.org/releases/theora/libtheora-1.1.1.tar.bz2
Trying a mirror...
==> Downloading https://ftp.osuosl.org/pub/xiph/releases/theora/libtheora-1.1.1.tar.bz2
Error: allegro: Failed to download resource "theora"
Download failed: Homebrew-installed `curl` is not installed for: https://downloads.xiph.org/releases/theora/libtheora-1.1.1.tar.bz2

Am I doing something incorrectly, or should I have done something before installing Allegro?

**Comment by carlocab** [comment]
What is the output of `brew config` and `brew doctor`?

**Reply by pyrolusite-d** [reply]
$ brew config
HOMEBREW_VERSION: 3.3.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5889b44a6031af91b14ee7c7805e64d1b10ad51c
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 088e2b86b2082af84f233969528d958703233c14
Core tap last commit: 4 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 10.0.0 build 1000
Git: 2.17.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.13.6-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

**Reply by carlocab** [reply]
You need to install `curl` and `ca-certificates` to be able to download the `theora` source, though this should have been done for you automatically. Did that not happen?

**Reply by pyrolusite-d** [reply]
I don't think so, just started using brew recently and I really don't know the nature of my issue. How do I install `curl` and `ca-certificates`?

**Reply by carlocab** [reply]
You can try `brew install curl ca-certificates`.

**Reply by pyrolusite-d** [reply]
Thank you, I think the issue is solved. <3

**Reply by mattneub** [reply]
Helpful, thank you. Apparently sometimes `brew` needs its _own_ curl, separate from any already installed `curl` available to it, and you have to ask for that to be installed and build explicitly before proceeding.

**Reply by carlocab** [reply]
> Apparently sometimes `brew` needs its _own_ curl, separate from any already installed `curl` available to it

Yes, sometimes. Usually when the already installed `curl` is too old. (Or doesn't live in `/bin` or `/usr/bin`, but there are workarounds to that. And it needs to not be too old too.)

> and you have to ask for that to be installed and build explicitly before proceeding.

We handle installing `curl` automatically when the system's is too old on macOS, but don't have a good way of working out whether the system's is too old on Linux currently. (Or maybe we do and we're not making good enough use of it.) PRs to improve this welcome.

**Comment by mattneub** [comment]
> We handle installing curl automatically when the system's is too old on macOS

Well, clearly not, since this is macOS (10.13, just like the original question). But it doesn't matter, as I said; saying that I needed to `brew install curl` and then start over was correct. — Just in case it's of interest, what I'm actually trying to do is `brew install ffmpeg`. This is longest install I've ever seen; it's been going on for four hours now, and it looks like it will take another four. I thought this whole config-make-install dance was just the sort of thing Homebrew was there to help bypass. But in any case the chokepoint in the original question was the same sort thing I encountered: `flac` and `theora` needed Homebrew's special `curl`. So thanks again for the hint.

**Reply by carlocab** [reply]
> Well, clearly not, since this is macOS.

Ah, well. That might be a bug. But it occurs on a version of macOS we don't test on, so bugs showing up are hard to avoid. Unless you're running Catalina or newer? But then you wouldn't need to build `ffmpeg` from source if you were.

@Bo98 did a bit of testing on older macOS versions recently and might know more about this.

**Reply by mattneub** [reply]
But no worries, time isn't an issue. 👍
